Add docs to a separate PR for review (#5153)
ROCm/rocm-libraries: projects/hipdnn/docs/reference/how-to-plugin.rst

.. meta::
  :description: 
  :keywords: hipDNN, ROCm, coding, naming

.. _how-to-plugin:

***********************************
hipDNN guides for plugin developers
***********************************

> These guides demonstrate how to use hipDNN's extended functionality, namely developing plugins for the platform.

Reviewer: ```suggestion
These guides demonstrate how to extend hipDNN's operation support using the hipDNN engine plugin interface.
```

Author: added

.. important::

>   These instructions are appropriate for plugin developers who have experience with hipDNN, and know how to build and compile a shared library (Linux) or DLL (Windows).

Reviewer: ```suggestion
  These instructions are appropriate for developers who have experience with hipDNN and are familiar with building and compiling shared libraries (Linux) or DLLs (Windows).
```

Author: Added
